src/EpisodeManagementService/GetEpisode/GetEpisode.cs

```diff
@@ -0,0 +1,61 @@
+using System.Net;
+using Microsoft.Azure.Functions.Worker;
+using Microsoft.Azure.Functions.Worker.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Configuration;
+using NHS.ServiceInsights.Common;
+
+namespace NHS.ServiceInsights.EpisodeManagementService
+{
+    public class GetEpisode
+    {
+        private readonly ILogger<GetEpisode> _logger;
+        private readonly IHttpRequestService _httpRequestService;
+        private readonly IConfiguration _configuration;
+
+        public GetEpisode(ILogger<GetEpisode> logger, IHttpRequestService httpRequestService, IConfiguration configuration)
+        {
+            _logger = logger;
+            _httpRequestService = httpRequestService;
+            _configuration = configuration;
+        }
+
+        [Function("GetEpisode")]
+        public async Task<HttpResponseData> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequestData req)
+        {
+            _logger.LogInformation("Request to retrieve episode information has been processed.");
+
+            string episodeId = req.Query["EpisodeId"];
+
+            if (string.IsNullOrEmpty(episodeId))
+            {
+                _logger.LogError("Please enter a valid Episode ID.");
+                var badRequestResponse = req.CreateResponse(HttpStatusCode.BadRequest);
+                return badRequestResponse;
+            }
+
+            var baseUrl = Environment.GetEnvironmentVariable("GetEpisodeUrl");
+            var url = $"{baseUrl}?EpisodeId={episodeId}";
+            _logger.LogInformation("Requesting URL: {Url}", url);
+
+            var response = await _httpRequestService.SendGet(url);
```

Comment:
The underlying GetAsync method call to HttpClient can potentially throw an exception so this line should be wrapped in a try catch block. If an exception happens we should return an InternalServerError and log the exception.